crates/iota-indexer/src/test_utils.rs

```diff
@@ -71,6 +73,7 @@ pub async fn start_test_indexer_impl<T: R2D2Connection + 'static>(
     data_ingestion_path: Option<PathBuf>,
     cancel: CancellationToken,
     new_database: Option<&str>,
+    sync_from_rest_api: bool,
```

Comment:
Why is this necessary?

Reply:
When enabling by default the REST API fetch most of e2e tests cases from localnet and graphQL were failing:

- https://github.com/iotaledger/iota/actions/runs/12030001477/job/33537011315
- https://github.com/iotaledger/iota/actions/runs/12030001477/job/33537488968

Reply:
Couldn't you use `data_ingestion_path = None` to fallback to the rest synchronization and keep the signature intact?

Reply:
@kodemartin [fixed](https://github.com/iotaledger/iota/pull/4173/commits/39384b270b251263cce1aaf40a2ad75dad4bc54d#diff-c1acc129a5b81dc72a6ce4d79513b7a2b3903911fc34c75681fc28feefb12b32R89-R92)